src/lcec_el7041_1000.c

```diff
@@ -0,0 +1,710 @@
+    return -EIO;
+  }
+  memset(hal_data, 0, sizeof(lcec_el7041_1000_data_t));
+  s->hal_data = hal_data;
+
+  // initialize sync info
+  s->sync_info = lcec_el7041_1000_syncs;
+
+  // initialize global data
+  hal_data->last_operational = 0;
+
+  // initialize pins
+  c = &hal_data->chans[0];
+
+  // initialize PDO entries
+
+#define pdo(...) LCEC_PDO_INIT(r, s->index, s->vid, s->pid, ##__VA_ARGS__)
+
+  pdo(0x7000, 0x01, &c->ena_latch_c_pdo_os,         &c->ena_latch_c_pdo_bp);
+  pdo(0x7000, 0x02, &c->ena_latch_ext_pos_pdo_os,   &c->ena_latch_ext_pos_pdo_bp);
+  pdo(0x7000, 0x03, &c->set_count_pdo_os,           &c->set_count_pdo_bp);
+  pdo(0x7000, 0x04, &c->ena_latch_ext_neg_pdo_os,   &c->ena_latch_ext_neg_pdo_bp);
+  pdo(0x7000, 0x11, &c->set_count_val_pdo_os,       NULL);
+
+  pdo(0x7010, 0x01, &c->dcm_ena_pdo_os,             &c->dcm_ena_pdo_bp);
+  pdo(0x7010, 0x02, &c->dcm_reset_pdo_os,           &c->dcm_reset_pdo_bp);
+  pdo(0x7010, 0x03, &c->dcm_reduce_torque_pdo_os,   &c->dcm_reduce_torque_pdo_bp);
+
+  pdo(0x7010, 0x21, &c->dcm_velo_pdo_os, NULL);
+
+  pdo(0x6000, 0x01, &c->latch_c_valid_pdo_os,   &c->latch_c_valid_pdo_bp);
+  pdo(0x6000, 0x02, &c->latch_ext_valid_pdo_os, &c->latch_ext_valid_pdo_bp);
+  pdo(0x6000, 0x03, &c->set_count_done_pdo_os,  &c->set_count_done_pdo_bp);
+  pdo(0x6000, 0x04, &c->count_underflow_pdo_os, &c->count_overflow_pdo_bp);
+  pdo(0x6000, 0x05, &c->count_overflow_pdo_os,  &c->count_underflow_pdo_bp);
+  pdo(0x6000, 0x08, &c->expol_stall_pdo_os,     &c->expol_stall_pdo_bp);
+  pdo(0x6000, 0x09, &c->ina_pdo_os,             &c->ina_pdo_bp);
+  pdo(0x6000, 0x0a, &c->inb_pdo_os,             &c->inb_pdo_bp);
+  pdo(0x6000, 0x0b, &c->inc_pdo_os,             &c->inc_pdo_bp);
+  pdo(0x6000, 0x0d, &c->inext_pdo_os,           &c->inext_pdo_bp);
+  pdo(0x6000, 0x0e, &c->sync_err_pdo_os,        &c->sync_err_pdo_bp);
+  pdo(0x6000, 0x10, &c->tx_toggle_pdo_os,       &c->tx_toggle_pdo_bp);
+  pdo(0x6000, 0x11, &c->count_pdo_os, NULL);
+  pdo(0x6000, 0x12, &c->latch_pdo_os, NULL);
+
+  pdo(0x6010, 0x01, &c->dcm_ready_to_enable_pdo_os, &c->dcm_ready_to_enable_pdo_bp);
+  pdo(0x6010, 0x02, &c->dcm_ready_pdo_os,           &c->dcm_ready_pdo_bp);
+  pdo(0x6010, 0x03, &c->dcm_warning_pdo_os,         &c->dcm_warning_pdo_bp);
+  pdo(0x6010, 0x04, &c->dcm_error_pdo_os,           &c->dcm_error_pdo_bp);
+  pdo(0x6010, 0x05, &c->dcm_move_pos_pdo_os,        &c->dcm_move_pos_pdo_bp);
+  pdo(0x6010, 0x06, &c->dcm_move_neg_pdo_os,        &c->dcm_move_neg_pdo_bp);
+  pdo(0x6010, 0x07, &c->dcm_torque_reduced_pdo_os,  &c->dcm_torque_reduced_pdo_bp);
+  pdo(0x6010, 0x0c, &c->dcm_din1_pdo_os,            &c->dcm_din1_pdo_bp);
+  pdo(0x6010, 0x0d, &c->dcm_din2_pdo_os,            &c->dcm_din2_pdo_bp);
+  pdo(0x6010, 0x0e, &c->dcm_sync_err_pdo_os,        &c->dcm_sync_err_pdo_bp);
+  pdo(0x6010, 0x10, &c->dcm_tx_toggle_pdo_os,       &c->dcm_tx_toggle_pdo_bp);
+
+#undef pdo
+
+#define msg_pin(format) \
```

Comment:
This is nice stuff. Maybe I will move that later to a mor generic place to use it in other modules.